dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskInstanceServiceImpl.java

```diff
@@ -193,7 +199,26 @@ public Map<String, Object> forceTaskSuccess(User loginUser, long projectCode, In
             putMsg(result, Status.TASK_INSTANCE_STATE_OPERATION_ERROR, taskInstanceId, task.getState().toString());
             return result;
         }
-
+        // Workflow enforcement succeeded
+        ProcessInstance processInstance = processService.findProcessInstanceDetailById(task.getProcessInstanceId());
```

Comment:
Suggest extract a method

Reply:
The update workflow and update task are in the same transaction, and this code is difficult to be universal. So this was unnecessary

Reply:
Nobody wants to read long functions.Write the function shorter, the shorter the better.